src/introductory-offer-banner/get-offer-message.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+export const getOfferMessage = ( {
+	formattedPrice,
+	intervalUnit,
+	intervalCount,
+}: {
+	formattedPrice: string;
+	intervalUnit: string;
+	intervalCount: string;
+} ): string => {
+	let message = '';
+	if ( intervalCount === '1' ) {
+		message = sprintf(
+			/* translators:  placeholders do not need to be translated */
+			__(
+				'Upgrade your plan for %(formattedPrice)s for your first %(intervalUnit)s',
+				'wc-calypso-bridge'
+			),
+			{
+				formattedPrice,
+				intervalUnit,
+			}
+		);
+	} else {
+		message = sprintf(
+			/* translators:  %(intervalUnit)s: one of day, month, and year, formattedPrice and intervalCount do not need to be translated. */
+			__(
+				'Upgrade your plan for %(formattedPrice)s for your first %(intervalCount)s %(intervalUnit)ss',
```

Comment:
AFAIK if we do it like this, `intervalUnit` is not translated. I think it could be like this in Spanish:

`Actualice su plan por $1 por su primer 3 months`

Thus I think we need to have all the possible interval units written in our copy here.

Reply:
@ilyasfoo  Thank you for pointing it out 👍  TIL.

Listed all possible interval units in https://github.com/Automattic/wc-calypso-bridge/pull/1397/commits/8196d18bdaa6c8a33d1d3c29c611520a2fb1680f